tck/src/main/java/org/eclipse/microprofile/config/tck/ArrayConverterTest.java

```diff
@@ -1654,120 +1658,161 @@ public void testInstantSetInjection() {
+        URL[] values = converterBean.getMyUrlArray();
+        Assert.assertEquals(values.length, 3);
+        assertURLArrayEquals(values);
         Assert.assertEquals(converterBean.getMySingleUrlArray().length, 1);
-        Assert.assertEquals(converterBean.getMySingleUrlArray(), new URL[]{ new URL("http://microprofile.io") });
+        Assert.assertEquals(converterBean.getMySingleUrlArray()[0].toURI(), new URL[]{ new URL("http://microprofile.io") }[0].toURI());
     }
 
     @Test
-    public void testURLListInjection() throws MalformedURLException {
-        Assert.assertEquals(converterBean.getMyUrlList().size(), 3);
-        Assert.assertEquals(converterBean.getMyUrlList(), Arrays.asList(
-            new URL("http://microprofile.io"),
+    public void testURLListInjection() throws MalformedURLException, URISyntaxException {
+        List<URL> values = converterBean.getMyUrlList();
+        Assert.assertEquals(values.size(), 3);
+        assertURLListEquals(values);
+        Assert.assertEquals(converterBean.getMySingleUrlList().size(), 1);
+        Assert.assertEquals(converterBean.getMySingleUrlList().get(0).toURI(), Arrays.asList(new URL("http://microprofile.io")).get(0).toURI());
+    }
+
+    private boolean assertURLSetEquals(Set<URL> valueSet, Set<URL> expectedURLSet) throws MalformedURLException, URISyntaxException {
+        if (valueSet.size()!= expectedURLSet.size()) {
+            return false;
+        }
+        Iterator<URL> it = valueSet.iterator();
+        boolean isEquals = true;
+        while (it.hasNext()) {
+            boolean found = false;
+            URL url = it.next();
+            for (URL thisURL : expectedURLSet) {  
+                if (thisURL.toURI().equals(url.toURI())) {
+                    found = true;
+                    break;
+                }
+            }
+            if (!found) {
+                isEquals = false;
+                break;
+            }
+        }
+        return isEquals;
+            
+    }
+    @Test
+    public void testURLSetInjection() throws MalformedURLException, URISyntaxException {
+        Set<URL> values = converterBean.getMyUrlSet();
+        Assert.assertEquals(values.size(), 2);
+        Set<URL> expectedURLSet = new LinkedHashSet<>(Arrays.asList(
             new URL("http://openliberty.io"),
             new URL("http://microprofile.io")));
-        Assert.assertEquals(converterBean.getMySingleUrlList().size(), 1);
-        Assert.assertEquals(converterBean.getMySingleUrlList(), Arrays.asList(new URL("http://microprofile.io")));
-    }
-
-    @Test
-    public void testURLSetInjection() throws MalformedURLException {
-        Assert.assertEquals(converterBean.getMyUrlSet().size(), 2);
-        Assert.assertEquals(converterBean.getMyUrlSet(), new LinkedHashSet<>(Arrays.asList(
-                new URL("http://openliberty.io"),
-                new URL("http://microprofile.io"))));
+        Assert.assertTrue(assertURLSetEquals(values, expectedURLSet));
```

Comment:
As sets are used here I wonder if the lists and arrays used before are intentionally expecting an order or if those really should compare sets as well. Can say but wanted to point it out. 
Using a `LinkedHashSet` also bags the question if order does matter in which case these maybe should be compared as lists? Or use the default `HashSet` to leave no question that these are sets and order is not important.

Reply:
not really. We should not force ordering for sets.

Reply:
> We should not force ordering for sets.

Then why use a `LinkedHashSet`?